File: Data Structures & Algorithms/min-cost-climbing-stairs/submission-1.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        if len(cost) == 0:
            return 0

        if len(cost) <= 2:
            return min(cost)
        
        leftMost = cost[0]
        left = cost[1]

        for i in range(2, len(cost)):
            temp = left
            left = cost[i] + min(temp, leftMost)
            leftMost = temp

        return min(left, leftMost)

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array-ii/submission-1.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int n = nums.size();
        if (n == 1)
            return n;

        int i = 0;
        int j = 0;
        while (j < n)
        {
            int count{1};
            while (j + 1 < n && nums[j] == nums[j + 1])
            {
                count++;
                ++j;
            }
            count = std::min(count, 2);
            for (int k = 0; k < count; ++k)
            {
                nums[i] = nums[j];
                i++;
            }
            ++j;
        }
        return i;
    }
};